controllers/eventhandlers/grpcroute.go

```diff
@@ -0,0 +1,53 @@
+package eventhandlers
+
+import (
+	"context"
+
+	"sigs.k8s.io/gateway-api/apis/v1alpha2"
+
+	"github.com/aws/aws-application-networking-k8s/pkg/k8s"
+	"github.com/aws/aws-application-networking-k8s/pkg/utils/gwlog"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+type grpcRouteEventHandler struct {
+	log    gwlog.Logger
+	client client.Client
+	mapper *resourceMapper
+}
+
+func NewGRPCRouteEventHandler(log gwlog.Logger, client client.Client) *grpcRouteEventHandler {
+	return &grpcRouteEventHandler{log: log, client: client,
+		mapper: &resourceMapper{log: log, client: client}}
+}
+
+func (h *grpcRouteEventHandler) MapToIAMAuthPolicies() handler.EventHandler {
```

Comment:
I would like to have a test code for this, easier way to test is extract logic into method level so that you can refer to it outside